pymysql/cursors.py

```diff
@@ -158,6 +159,8 @@ def executemany(self, query, args):
 
         m = RE_INSERT_VALUES.match(query)
         if m:
+            if self.max_stmt_length == Cursor.max_stmt_length:
+                self.max_stmt_length = self.connection.get_system_variable("max_allowed_packet") - self.packet_header_size
```

Comment:
Using `max_allowed_packet` may eat too many memory.
We should have more reasonable soft limit.  (I think 1MB is reasonable).